apollo-router/src/plugins/authentication/subgraph.rs

```diff
@@ -509,21 +546,22 @@ mod test {
         mock.expect_call()
             .times(1)
             .withf(|request| {
-                let authorization_regex = Regex::new(r"AWS4-HMAC-SHA256 Credential=id/\d{8}/us-east-1/s3/aws4_request, SignedHeaders=content-length;content-type;host;x-amz-content-sha256;x-amz-date, Signature=[a-f0-9]{64}").unwrap();
-                let authorization_header_str = request.subgraph_request.headers().get("authorization").unwrap().to_str().unwrap();
+                let http_request = get_signed_request(request);
+                let authorization_regex = Regex::new(r"AWS4-HMAC-SHA256 Credential=id/\d{8}/us-east-1/s3/aws4_request, SignedHeaders=host;x-amz-content-sha256;x-amz-date, Signature=[a-f0-9]{64}").unwrap();
```

Comment:
cc @nmoutschen @BlenderDude I had to change the signedheaders value because we decided to only use the ones that are mandatory (sigv4 is not happy with the connection header, and maybe others).

I'd like to be 100% sure it's not gonna be a problem eventually. What do you both think?